### PR TITLE
Allow image pasting in plain mode in RTE

### PR DESCRIPTION
**Pull request by alunturner**
This PR brings the plain text mode of the RTE in line with the rich text mode, in that pasting with an image on the clipboard now opens a modal giving the user the option to upload an image. Example below.

Element__2____Alun_Turner

The PR reuses as much of the logic used by the rich text mode as possible. This meant extracting the existing functions to a sensible place.

This PR:

- extracts the functions that are common to the rich and plain text modes to a `utils` file
- uses those functions for the plain text editor
- adds tests

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Allow image pasting in plain mode in RTE ([\#11056](https://github.com/matrix-org/matrix-react-sdk/pull/11056)). Contributed by @alunturner.

### CODE REVIEW

File: test/components/views/rooms/wysiwyg_composer/components/PlainTextComposer-test.tsx [hunk 1]
```diff
@@ -290,4 +290,16 @@ describe("PlainTextComposer", () => {
...
+        customRender();
+
+        const textBox = screen.getByRole("textbox");
+
+        await userEvent.click(textBox);
+        await userEvent.type(textBox, "hello");
+        await userEvent.paste(" world");
+
+        expect(textBox).toHaveTextContent("hello world");
+    });
```
Comment (alunturner) [inline test/components/views/rooms/wysiwyg_composer/components/PlainTextComposer-test.tsx:304]: Test to increase (non-existant) pasting coverage in general in this component

**Comment by alunturner** [comment]
Please see the rich text version of this implementation for discussion as to why a `beforeinput` listener is required for this work: https://github.com/matrix-org/matrix-react-sdk/pull/11049

**Review by justjanne — APPROVED** [review]

**Comment by alunturner** [comment]
Interesting, thanks for the feedback. Where did you test it?

**Review by justjanne — APPROVED** [review]

**Comment by justjanne** [comment]
@alunturner I just did a lot more testing, and it turns out firefox only exposes images in the clipboard paste event that were also copied from within firefox. So if I go onto another website, right-click, copy, I can paste it in element. If I right-click a file in my file explorer, I can paste it in other apps, but not in firefox. So the issue I experienced was caused by my environment.

**Pull request merged**